### DISCUSSION TITLE
Custom Tracker Integration for College Softball specifcally Oklahoma

### DISCUSSION THREAD

**Discussion by bcayton** [vasqued2/ha-teamtracker#138] [Q&A]
Greetings All, 

I am trying to figure out if it is possible to create a custom team tracker for women's college softball specifically Oklahoma. My wife wants me to set up a game reminder automation like I have done for football and basketball. I have read the information on the wiki however with ESPNs API DEV center being taken down its making it hard to figure out the required information. Any assistance is much appreciated.

**Comment by vasqued2** [comment]
This should work

```
sport_path:  baseball
league_path:  college-softball
```

**Reply by bcayton** [reply]
Thanks but when I put it in I’m still getting API error no data returned.

**Comment by vasqued2** [comment]
You have something wrong if you are getting no data returned.  This works for me.  Using conference_id will keep the API from hitting the 25 game limit.  Big 12 is 45.

```
  - platform: teamtracker
    league_id: XXX
    team_id: OSU
    conference_id: 49
    sport_path: baseball
    league_path: college-softball
    name: test_tt_all_test32
```

**Reply by bcayton** [reply]
Thanks David I appreciate all your support keeping this integration up. I think I finally got it. I took me a couple of tries I must have been fat fingering something

**Comment by vasqued2** [comment]
Good news!

**Comment by vasqued2** [comment]
Is there some sort of run rule thing for softball?  I was tracking the Iowa State game today and noticed I wasn't getting updates.  I assumed it was because updates weren't consistent for women's softball but not it shows as Final/5, which I'm wondering if it means the game was called after 5 innings?

**Comment by bcayton** [comment]
So I’m not the expert but I asked my wife who played in college, and yes there is a “run rule”. If a team is up 8 runs after 5 innings the game can officially be called.

**Comment by vasqued2** [comment]
That must have been what happened.  Congrats on the win.

**Comment by bcayton** [comment]
@vasqued2 Can you take a look at this again for me? Looks like ESPN has changes their API's and I can't figure out what the new one is for college softball. I have updated the conference number to reflect 32 for SEC Softball. However, the tracker keeps returning a "NOT_FOUND". I really don't understand ESPN's API structure for non "major" sports. Thanks in advance! I love this integration.

**Comment by vasqued2** [comment]
@bcayton,

Sorry, I just noticed this.  It looks like ESPN stopped using conference ID's for minor sports.  This caused a lot of minor college sports to hit the API limit and results not to show up. I just made an update in v0.14.9 to handle Lacrosse but it should work for softball too.

Lmk if you still have problems and I'll keep looking.  I can't find an upcoming game to check now.

**Comment by bcayton** [comment]
@vasqued2 

I played with it and I still can’t get it to work. I’m not sure if it’s me, ESPN, the integration or a little of all three. 

I’ve tried using the following: 

platform: teamtracker
    league_id: XXX
    team_id: OU
    conference_id: 32 or left blank
    sport_path: baseball
    league_path: college-softball
    name: Oklahoma Sooner Softball 

it’s still coming back as the API is not valid. 

They next play tomorrow April 25th at 5:00 pm. 

Thank you so much for the help! I know it must suck when you get everything working good and then ESPN changes their API structure.

**Reply by bcayton** [reply]
Update: I think it got it working after playing some more. We will see if it updates for the game tomorrow.

**Reply by vasqued2** [reply]
It should work.  You may have to drop the Conference ID but the expanded limit should pull it in.  Prior to the recent change, you might have gotten it if or it may have been truncated due to the limit being hit, purely luck.

**Comment by bcayton** [comment]
@vasqued2 

Random question, do you know what time ESPN updates their API data? I went in this morning to check and see if team tracker updated correctly and it’s still showing data from yesterday 04/24 vice today 04/25.

**Comment by bcayton** [comment]
The integration is showing its returning cached data. 

“ Cached data: No competition scheduled for 'OU' between 2025-04-25T01:00Z and 2025-04-25T01:00Z”

**Comment by vasqued2** [comment]
The OU softball sensor is working for me and is providing the results of today's game.  I did have to make sure there was no `conference_id` set. I see based on an old sensor I used to test this originally I used `conference_id` = 45. That sensor no longer works so ESPN definitely changed something up w/ the conference IDs for softball. If it's not working for you, make sure the API URL matches the screenshot below.

Cached data indicates that you have more than one sensor set up for sofball.  The first sensor calls the API and the second will use the cached data until it times out (after 5 seconds or 10/20 minutes I forget which). It just is something that keeps the API from getting hammered if you already called it recently for another sensor.

It should be smart enough to know not to try to reuse one with a bad conference_id but if you think that is happening, I will look into it.

![image](https://github.com/user-attachments/assets/dfbb98a0-9371-461c-8973-0aed0294bd34)

![image](https://github.com/user-attachments/assets/c257e932-9203-4cce-b5ac-3bac1f9ecbb4)

**Reply by bcayton** [reply]
Thanks @vasqued2 I really appreciate all the support. It eventually updated for me, but it definitely took longer than 20 minutes. Regardless it’s working now and I really appreciate all your hard work.
